akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala

```diff
@@ -2024,6 +2024,47 @@ final class Flow[In, Out, Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends Gr
   def prefixAndTail(n: Int): javadsl.Flow[In, akka.japi.Pair[java.util.List[Out], javadsl.Source[Out, NotUsed]], Mat] =
     new Flow(delegate.prefixAndTail(n).map { case (taken, tail) => akka.japi.Pair(taken.asJava, tail.asJava) })
 
+  /**
+   * Takes up to `n` elements from the stream (less than `n` only if the upstream completes before emitting `n` elements),
+   * then apply `f` on these elements in order to obtain a flow, this flow is then materialized and the rest of the input is processed by this flow (similar to via).
+   * This method returns a flow consuming the rest of the stream producing the materialized flow's output.
+   *
+   * '''Emits when''' the materialized flow emits.
+   *  Notice the first `n` elements are buffered internally before materializing the flow and connecting it to the rest of the upstream - producing elements at its own discretion (might 'swallow' or multiply elements).
+   *
+   * '''Backpressures when''' downstream backpressures
```

Comment:
Is that still right, though? Maybe something like

```suggestion
   * '''Backpressures''' after the prefix has been collected when the materialized flow backpressures. The prefix is collected eagerly without applying backpressure.
```

That then needs to be propagated. You could also do that change in a follow-up PR, so that this one can be already merged.

Reply:
this stage actually doesn't back pressures until materializing the flow,  it will keep consuming as long as it have to fill the prefix (very similar to buffer in this respect).
it will only start backpressuring  once the flow is materialized and backpressures on its own.

so basically I think the wording is right, it might make sense to add some more wording in the docs about the detailed behaviour of this operator.

Reply:
Currently it says

> '''Backpressures when''' downstream backpressures

which doesn't seem correct.